chef/cookbooks/ceph/recipes/osd.rb

```diff
@@ -103,7 +110,15 @@
         Chef::Log.info("Ceph: Claimed #{d.name}")
         device = {}
         dev_name = d.name.gsub("/dev/", "")
-        if node["ceph"]["osd"]["journal_devices"].include?(d.name) || (node[:block_device][dev_name]["rotational"] == "0" && node["ceph"]["osd"]["use_ssd_for_journal"])
+        if node["ceph"]["osd"]["journal_devices"].include?(d.name) ||
+            (node[:block_device][dev_name]["rotational"] == "0" &&
+             node["ceph"]["osd"]["use_ssd_for_journal"] &&
+             node["ceph"]["disk_mode"] == "all")
```

Comment:
Hm, I agree, your approach would catch this case and even if it might be uncommon one, it's no need to ignore it.